libcxx/test/support/count_new.h

```diff
@@ -16,6 +16,8 @@
 #include <new>
 #include <type_traits>
 
+#include <__memory/aligned_alloc.h> // reuse aligned allocation helper
```

Comment:
We don't want to reuse libc++ internal functionality in the test suite. One reason is that our test suite is used by other implementations of the C++ stdlib that are not libc++, and those don't have `__memory/aligned_alloc.h`. Another reason is that we should avoid relying on library functionality that we're testing *in the implementation of the test suite*, because that's kinda circular.

Reply:
On the other hand, writing a separate set of ifdefs in two places intended to do the same job is a recipe for updating one and forgetting to update the other, so they get out of sync!

What's your alternative? I accept that we can't depend on a libc++ internal header if we're not always testing libc++ (which I didn't know, thanks). Can we move the huge pile of "guess which aligned allocation API to use" ifdefs to a place where both the tests and libc++ can include it?

Reply:
Frankly, I would just duplicate a bit of logic. I don't think it makes sense to create a lot of complexity just to solve this one instance of code duplication.